### PR TITLE
add util tests

### PR DESCRIPTION
**Pull request by achamas-playco**
> NOTE: 📣 This PR has 2 skipped test, they each have a corresponding investigation ticket
> PR is ready for review and merge

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4ddeb0f</samp>

This pull request adds and improves several features and functions related to rendering textures from canvases and videos, and measuring and trimming canvases. It also adds and updates tests for the new and existing code.

### CODE REVIEW

File: src/utils/canvas/BoundingBox.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/**
+ * A rectangular bounding box describing the boundary of an area.
+ * @since 7.1.0
```
Comment (Zyie) [inline src/utils/canvas/BoundingBox.ts:3]:
```suggestion
 * @since 7.1.0
 * @memberof utils
```

File: src/utils/canvas/getCanvasBoundingBox.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    return true;
+}
+
+/**
+ * Measuring the bounds of a canvas' visible (non-transparent) pixels.
+ * @param {ICanvas} canvas - The canvas to measure.
+ * @returns {BoundingBox} The bounding box of the canvas' visible pixels.
+ * @since 7.1.0
```
Comment (Zyie) [inline src/utils/canvas/getCanvasBoundingBox.ts:31]:
```suggestion
 * @since 7.1.0
 * @memberof utils
```

File: src/utils/canvas/trimCanvas.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { getCanvasBoundingBox } from './getCanvasBoundingBox';
+
+import type { ICanvas } from '../../environment/canvas/ICanvas';
+
+/**
+ * Trim transparent borders from a canvas.
+ * @param {ICanvas} canvas - The canvas to trim.
+ * @returns The trimmed canvas data.
```
Comment (Zyie) [inline src/utils/canvas/trimCanvas.ts:8]:
```suggestion
 * @returns The trimmed canvas data.
 * @memberof utils
```

File: src/rendering/renderers/shared/renderTarget/RenderTarget.ts [hunk 4]
```diff
@@ -185,6 +193,14 @@ export class RenderTarget
 
     public destroy()
     {
-        throw new Error('Method not implemented.');
+        for (const colorTexture of this.colorTextures)
+        {
+            colorTexture.destroy();
+        }
```
Comment (Zyie) [inline src/rendering/renderers/shared/renderTarget/RenderTarget.ts:199]: I think we also need to destroy the `depthTexture` as well if it exists

Comment (achamas-playco) [reply src/rendering/renderers/shared/renderTarget/RenderTarget.ts:199]: Done, thanks!

**Comment by Zyie** [comment]
@achamas-playco can you take a look at the merge conflicts?

**Comment by achamas-playco** [comment]
> @achamas-playco can you take a look at the merge conflicts?

Will do, though there's been some refactors on `RenderTarget` since this branch started and my initial attempts at rebasing were not simple and seem to have some issues, so I'll need to look at it further. Will update when completed.

**Comment by achamas-playco** [comment]
@Zyie has recreated this branch and solved the issues, thanks Sean!

**Pull request closed without merging**